examples/Titanic/titanic_captum_interpret.py

```diff
@@ -0,0 +1,375 @@
+"""
+Getting started with Captum - Titanic Data Analysis
+"""
+# Initial imports
+import numpy as np
+import torch
+from captum.attr import IntegratedGradients
+from captum.attr import LayerConductance
+from captum.attr import NeuronConductance
+import matplotlib.pyplot as plt
+from scipy import stats
+import pandas as pd
+import mlflow
+from prettytable import PrettyTable
+from sklearn.model_selection import train_test_split
+import os
+from argparse import ArgumentParser
+import torch.nn as nn
+
+mlflow.start_run(run_name="Titanic_Captum_mlflow")
+
+
+def get_titanic():
+    """
+    we now preprocess the data by converting some categorical features such as
+    gender, location of embarcation, and passenger class into one-hot encodings
+    We also remove some features that are more difficult to analyze
+    After processing, the features we have are:
+    Age: Passenger Age
+    Sibsp: Number of Siblings / Spouses Aboard
+    Parch: Number of Parents / Children Aboard
+    Fare: Fare Amount Paid in British Pounds
+    Female: Binary variable indicating whether passenger is female
+    Male: Binary variable indicating whether passenger is male
+    EmbarkC : Binary var indicating whether passenger embarked @ Cherbourg
+    EmbarkQ : Binary var indicating whether passenger embarked @ Queenstown
+    EmbarkS : Binary var indicating whether passenger embarked @ Southampton
+    Class1 : Binary var indicating whether passenger was in first class
+    Class2 : Binary var indicating whether passenger was in second class
+    Class3 : Binary var indicating whether passenger was in third class
+    url = "https://biostat.app.vumc.org/wiki/pub/Main/DataSets/titanic3.csv"
+    """
+    url = "https://biostat.app.vumc.org/wiki/pub/Main/DataSets/titanic3.csv"
+    titanic_data = pd.read_csv(url)
+    titanic_data = pd.concat(
+        [
+            titanic_data,
+            pd.get_dummies(titanic_data["sex"]),
+            pd.get_dummies(titanic_data["embarked"], prefix="embark"),
+            pd.get_dummies(titanic_data["pclass"], prefix="class"),
+        ],
+        axis=1,
+    )
+
+    titanic_data["age"] = titanic_data["age"].fillna(titanic_data["age"].mean())
+    titanic_data["fare"] = titanic_data["fare"].fillna(titanic_data["fare"].mean())
+    titanic_data = titanic_data.drop(
+        [
+            "name",
+            "ticket",
+            "cabin",
+            "boat",
+            "body",
+            "home.dest",
+            "sex",
+            "embarked",
+            "pclass",
+        ],
+        axis=1,
+    )
+    return titanic_data
+
+
+torch.manual_seed(1)  # Set seed for reproducibility.
+
+
+class TitanicSimpleNNModel(nn.Module):
```

Comment:
```suggestion

```
This class already available as part of model file. Please remove the duplicate classes